tests/trace_server/test_evaluation_execution_api.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+
+from tests.trace_server.completions_util import with_simple_mock_litellm_completion
+from weave.trace.refs import ObjectRef
+from weave.trace_server.execution_runner.run_as_user import with_client_bound_to_project
+from weave.trace_server.execution_runner.user_scripts.run_model import run_model
+from weave.trace_server.trace_server_interface import (
+    CallsQueryReq,
+    ObjCreateReq,
+    ObjQueryReq,
+    RunModelReq,
+    RunModelRes,
+    TraceServerInterface,
+)
+
+
+@pytest.mark.asyncio
+async def test_run_model(ch_only_trace_server: TraceServerInterface, client_creator):
+    """
+    Test the run_model API endpoint with isolated execution.
+
+    This test verifies that:
+    1. Models can be created and executed through the run_model API
+    2. Model execution is properly traced (creates call records)
+    3. Different projects are properly isolated from each other
+
+    The test creates models in two different projects and executes them,
+    ensuring that calls and objects are properly scoped to each project.
+    """
+    entity = "shawn"
+
+    def create_model(entity: str, project: str) -> str:
+        project_id = f"{entity}/{project}"
+        model_object_id = "test_run_model"
+        llm_model_val = {
+            "llm_model_id": "gpt-4o-mini",
+            "default_params": {
+                "messages_template": [
+                    {"role": "system", "content": "You are a helpful assistant."},
+                ],
+                "response_format": "text",
+            },
+        }
+        model_create_res = ch_only_trace_server.obj_create(
+            ObjCreateReq.model_validate(
+                {
+                    "obj": {
+                        "project_id": project_id,
+                        "object_id": model_object_id,
+                        "val": llm_model_val,
+                        "builtin_object_class": "LLMStructuredCompletionModel",
+                    }
+                }
+            )
+        )
+        model_digest = model_create_res.digest
+        return ObjectRef(
+            entity=entity,
+            project=project,
+            name=model_object_id,
+            _digest=model_digest,
+        ).uri()
+
+    async def run_model_harness(
+        entity: str,
+        project: str,
+        model_ref_uri: str,
+        user_input: str,
+        expected_output: str,
+        # Running with local mode true directly calls the underlying user script
+        # allowing for easier debugging of the user script
+        run_mode_local: bool = False,
+    ) -> RunModelRes:
+        project_id = f"{entity}/{project}"
+        req = RunModelReq.model_validate(
+            {
+                "project_id": project_id,
+                "model_ref": model_ref_uri,
+                "inputs": {"user_input": user_input},
+            }
+        )
```

Comment:
same here

Reply:
actually, i already don't like this. 
```
 ObjCreateReq.model_validate(
                {
                    "obj": {
                        "project_id": project_id,
                        "object_id": model_object_id,
                        "val": llm_model_val,
                        "builtin_object_class": "LLMStructuredCompletionModel",
                    }
                }
            )
        )
```

this requires importing `ObjSchemaForInsert` for obj value. I like treating this like a pure dictionary

Reply:
this makes it closer to server-level testing